src/pages/files.js

```diff
@@ -88,112 +130,204 @@ export default class Files extends React.Component {
     });
   };
 
+  handleBackMyStack = () => {
+    this.setState({
+      showModal: true,
+    });
+  };
+
+  handleContinueBacMyStack = () => {
+    // Get the key url of the file
+    const { savedFileUrl } = this.state;
+    const { openCollectiveRedirectUrl } = this.props;
+    const { hostname, protocol } = window.location;
+    const fileKey = savedFileUrl.Key.replace('.json', '');
+    const jsonUrl = `"${protocol}//${hostname}/${fileKey}/file/backing.json"`;
+    const contributionUrl = `${openCollectiveRedirectUrl}?data={"jsonUrl":${jsonUrl}}`;
```

Comment:
This is not properly constructed.

- a JSON is constructed with JSON.stringify()
- URL query parameters needs to be properly handled, for example with the URL API: https://developer.mozilla.org/en-US/docs/Web/API/URL